ClosedXML_Tests/Excel/Worksheets/XLSheetProtectionTests.cs

```diff
@@ -111,33 +171,14 @@ public void TestWorksheetWithSHA512Protection()
             {
                 var ws = wb.Worksheet("Sheet2");
                 Assert.IsTrue(ws.Protection.IsProtected);
-                // Protected with SHA-512 password - not yet supported.
-                Assert.Throws<ArgumentException>(() => ws.Unprotect());
-
-                // Protected with SHA-512 password - not yet supported.
-                Assert.Throws<ArgumentException>(() => ws.Unprotect("abc"));
-                Assert.IsTrue(ws.Protection.IsProtected);
-            }
-        }
-
-        [Test]
-        public void SetWorksheetProtectionCloning()
-        {
-            var ws1 = new XLWorkbook().AddWorksheet();
-            var ws2 = new XLWorkbook().AddWorksheet();
-
-            ws1.Protect("123")
-                .AllowElement(XLSheetProtectionElements.FormatEverything)
-                .DisallowElement(XLSheetProtectionElements.FormatCells);
-
-            Assert.AreEqual(XLSheetProtectionElements.FormatColumns | XLSheetProtectionElements.FormatRows | XLSheetProtectionElements.SelectEverything, ws1.Protection.AllowedElements);
 
-            ws2.Protection = ws1.Protection;
+                // Password required
+                Assert.Throws<InvalidOperationException>(() => ws.Unprotect());
 
-            Assert.IsFalse(ReferenceEquals(ws1.Protection, ws2.Protection));
-            Assert.IsTrue(ws2.Protection.IsProtected);
-            Assert.AreEqual(XLSheetProtectionElements.FormatColumns | XLSheetProtectionElements.FormatRows | XLSheetProtectionElements.SelectEverything, ws2.Protection.AllowedElements);
-            Assert.AreEqual((ws1.Protection as XLSheetProtection).PasswordHash, (ws2.Protection as XLSheetProtection).PasswordHash);
+                Assert.AreEqual(XLProtectionAlgorithm.SHA512, ws.Protection.Algorithm);
+                ws.Unprotect("abc");
+                Assert.IsFalse(ws.Protection.IsProtected);
+            }
```

Comment:
I would also add a couple of tests for hashing functions just to protect us for accidental modification of algorithms.

E.g.
`
            var hash = CryptographicAlgorithms.GetPasswordHash(XLProtectionAlgorithm.SHA512, "PASSWORDS", "salt", 100000);
            Assert.AreEqual("kqxXTHzC3J8l5UkuSpKkZOwB9mv4lnTCJeDgpazBNRHkETFHi/VYHMHy7uJhWrBF+h7L96K4rwwUF5lBrXQSWg==", hash);
`